lib/actual_dqt/client.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require "net/http"
+
+module ActualDqt
+  class Client
+    attr_reader :token
+
+    def initialize(token:)
+      @token = token
+    end
+
+    def get_record(trn:, birthdate:, nino: nil)
+      uri_for_record = uri(trn: trn, birthdate: birthdate, nino: nino)
+
+      request = Net::HTTP::Get.new(uri_for_record)
+      request["Authorization"] = "Bearer #{token}"
+      request["Ocp-Apim-Subscription-Key"] = subscription_key
+
+      response = Net::HTTP.start(uri_for_record.hostname, uri_for_record.port, use_ssl: true, read_timeout: 20) do |http|
+        http.request(request)
+      end
+
+      if response.code == "200"
+        translate_hash(JSON.parse(response.body))
+      end
+    end
+
+  private
+
+    def translate_hash(hash)
+      hash.deep_transform_values do |value|
+        if value =~ /^\d{4}-\d{2}-\d{2}/
+          Date.parse(value)
+        else
+          value
+        end
+      end
+    end
```

Comment:
Do we know the names of the fields that this should apply to? Would it be safer to parse based on that, rather than field shape?

Reply:
I did consider that. From the data I've seen i don't expect anything shaped like a date that is not a date so I think the above is fine. I also didn't see a particularly nice way of doing the above for each key with a given name in a nested hash which may also turn out to be optional.

Reply:
Fair enough